client/src/services/cryptpad.ts

```diff
@@ -164,7 +180,49 @@ export class Cryptpad {
       throw new CryptpadOpenError(CryptpadErrorCode.DocumentTypeNotEnabled, config.documentType);
     }
 
-    (window as any).CryptPadAPI(this.containerElement.id, { ...config });
+    // Set up a loading timeout (30 seconds) to detect stuck/corrupted files
+    const LOADING_TIMEOUT_MS = 30000;
+    this.clearLoadingTimeout();
+
+    // Create a promise that will be resolved when CryptPad successfully loads
+    let resolveLoading: (() => void) | null = null;
+    const loadingComplete = new Promise<void>((resolve) => {
+      resolveLoading = resolve;
+    });
+
+    const timeoutPromise = new Promise<void>((_, reject) => {
+      this.loadingTimeoutId = window.setTimeout(() => {
+        window.electronAPI.log('error', 'CryptPad loading timeout - file may be corrupted or too large');
+        reject(new Error('CryptPad loading timeout - the file may be corrupted, too large, or the server is not responding'));
+      }, LOADING_TIMEOUT_MS);
+    });
+
+    // Wrap the original onReady callback to signal successful loading
+    const originalOnReady = config.events.onReady;
+    config.events.onReady = () => {
+      if (resolveLoading) resolveLoading();
+      if (originalOnReady) originalOnReady();
+    };
+
+    // Store config globally so CryptPad customization can access the onError callback
+    (window as any).cryptpadConfig = config;
+
+    try {
+      // Start CryptPad API (doesn't wait for loading to complete)
+      void (window as any).CryptPadAPI(this.containerElement.id, { ...config });
+
+      // Race between successful loading (onReady callback) and timeout
+      await Promise.race([loadingComplete, timeoutPromise]);
+
+      // If we reach here, onReady was called before timeout
+      this.clearLoadingTimeout();
+    } catch (error) {
+      this.clearLoadingTimeout();
+      throw error;
```

Comment:
From what I understand, this error is coming straight from cryptpad, it should be converted to a `CryptpadOpenError`. This way the caller doesn't have to wonder about what type of exception it receives and we can add a bit of abstraction.